galsim/photon_array.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2012-2016 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+import numpy as np
+from ._galsim import PhotonArray
+import galsim
+
+class FRatioAngles(object):
+    """A surface-layer operator that assigns photon directions based on the f/ratio and
+    obscuration.
+
+    Assigns arrival directions at the focal plane for photons, drawing from a uniform
+    brightness distribution between the obscuration angle and the angle of the FOV defined
+    by the f/ratio of the telescope.  The angles are expressed in terms of slopes dx/dz
+    and dy/dz.
+
+    @param fratio           The f-ratio of the telescope (e.g. 1.2 for LSST)
+    @param obscuration      Linear dimension of central obscuration as fraction of aperture
+                            linear dimension. [0., 1.).  [default: 0.0]
+    @param rng              A random number generator to use or None, in which case an rng
+                            will be automatically constructed for you. [default: None]
+    """
+    def __init__(self, fratio, obscuration=0.0, rng=None):
+
+        if fratio < 0:
+            raise ValueError("The f-ratio must be positive.")
+        if obscuration < 0 or obscuration >= 1:
+            raise ValueError("The obscuration fraction must be between 0 and 1.")
```

Comment:
Probably worth adding some assert_raises tests to check these lines work properly.  (They aren't required for our coverage report, but they're worth covering anyway.).  Something like the following:
```
try:
    np.testing.assert_raises(ValueError, FRatioAngles, fratio=-0.3)
    np.testing.assert_raises(ValueError, FRatioAngles, fratio=1.2, obscuration=-0.3)
    np.testing.assert_raises(ValueError, FRatioAngles, fratio=1.2, obscuration=1.0)
    np.testing.assert_raises(ValueError, FRatioAngles, fratio=1.2, obscuration=1.9)
except ImportError:
    pass
```
The try/except wrapper is because these tests technically require nose, which we don't have as an official dependency, so we don't require the user to have it installed.